### ISSUE TITLE
Ability to export chats using a time range (instead of number of messages)

### ISSUE THREAD

**Issue by nekohayo** [element-hq/element-web#26926]
### Your use case

The basic chat log exporting feature implemented in https://github.com/element-hq/element-web/issues/2630 works, but I find it very limiting when you can only specify these types of ranges:
* "Current timeline": it is unclear what this actually does. Like, what constitutes the beginning and end of "current"?
* "From the beginning": very rarely do you actually want to do this
* "Specify a number of messages": I never know exactly what amount of messages I want to export, because I'm not going to start counting them myself... so in practice I always end up vastly underestimating or vastly overestimating.

I would like to be able to export using a range of dates (and/or hours). Ex: from `YYYY-MM-DD HH:MM` to `YYYY-MM-DD`, or "last 6 hours", etc.

**Comment by t3chguy** [comment]
Related

**Comment by nekohayo** [comment]
Indeed, I was not able  to find that other ticket because it was titled to generically; that said, it also talks about other features / a wider scope. This ticket here is a subset of it that could potentially be implemented independently as a first step.

**Comment by fran-f** [comment]
Getting close to the end of the year, this issue comes back to my mind. Having a way to download the past 12 month of messages would make incremental backups possible.

There is already a bit of code commented out here:
https://github.com/element-hq/element-web/blob/18ef9753868248752b2280c2f009870d8df8a254/src/utils/exportUtils/Exporter.ts#L167-L171

Would be great to have a start date selector in the export dialog, without the complex interface discussed in #19310.

**Comment by t3chguy** [comment]
Contributions welcome

**Comment by fran-f** [comment]
I'll start with a mock-up!

![image](https://github.com/user-attachments/assets/9be4cd97-6e0c-4b28-b421-1d47014ba0b9)

**Comment by fran-f** [comment]
Alternatively, with a date range. The assumed time would be `00:00:00` for the start date, and `23:59:59` for the end date.

![image](https://github.com/user-attachments/assets/b84b95a9-4ef3-4feb-acbf-c53c9851998a)

**Comment by t3chguy** [comment]
Keep in mind timestamps may appear in pages out of order, as servers over federation may not have synchronised clocks, heterogenous distributed systems and all that

**Comment by fran-f** [comment]
Thanks for pointing that out. Then a single event is not enough to signal the boundary of a range, but we should probably check "a few minutes further" for safety?
